### PR TITLE
fix: use python module path in pymodules

### PR DESCRIPTION
**Pull request by numb3r3**
Goals:

To fix the issue when using python module path in the meta `py_modules`, such as:

```python
from jina import Flow, Document

f = Flow().add(uses='!CLIPEncoder', uses_meta={'py_modules': ['clip_server.executors.clip_torch']})

with f:
    r = f.post(on='/', inputs=Document(text='hello world'))
    print(r.embeddings)
```
raise the following error:
```
CRITI… executor0/rep-0@26144 can not load the executor from  [07/15/22 16:09:18]
       CLIPEncoder                                                              
ERROR  executor0/rep-0@26144 TypeError('stat: path should be [07/15/22 16:09:18]
       string, bytes, os.PathLike or integer, not NoneType')                    
       during <class                                                            
       'jina.serve.runtimes.worker.WorkerRuntime'>                              
       initialization                                                           
        add "--quiet-error" to suppress the exception                           
       details                                                                  
       ╭──────── Traceback (most recent call last) ────────╮                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/orchest… │                    
       │ in run                                            │                    
       │                                                   │                    
       │    71 │                                           │                    
       │    72 │   try:                                    │                    
       │    73 │   │   _set_envs()                         │                    
       │ ❱  74 │   │   runtime = runtime_cls(              │                    
       │    75 │   │   │   args=args,                      │                    
       │    76 │   │   )                                   │                    
       │    77 │   except Exception as ex:                 │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/serve/r… │                    
       │ in __init__                                       │                    
       │                                                   │                    
       │    28 │   │   :param kwargs: keyword args         │                    
       │    29 │   │   """                                 │                    
       │    30 │   │   self._health_servicer = health.Heal │                    
       │ ❱  31 │   │   super().__init__(args, **kwargs)    │                    
       │    32 │                                           │                    
       │    33 │   async def async_setup(self):            │                    
       │    34 │   │   """                                 │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/serve/r… │                    
       │ in __init__                                       │                    
       │                                                   │                    
       │    63 │   │   │   )                               │                    
       │    64 │   │                                       │                    
       │    65 │   │   self._setup_monitoring()            │                    
       │ ❱  66 │   │   self._loop.run_until_complete(self. │                    
       │    67 │                                           │                    
       │    68 │   def run_forever(self):                  │                    
       │    69 │   │   """                                 │                    
       │                                                   │                    
       │ /Users/fengwang/.pyenv/versions/3.8.6/Library/Fr… │                    
       │ in run_until_complete                             │                    
       │                                                   │                    
       │    613 │   │   if not future.done():              │                    
       │    614 │   │   │   raise RuntimeError('Event loop │                    
       │    615 │   │                                      │                    
       │ ❱  616 │   │   return future.result()             │                    
       │    617 │                                          │                    
       │    618 │   def stop(self):                        │                    
       │    619 │   │   """Stop running the event loop.    │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/serve/r… │                    
       │ in async_setup                                    │                    
       │                                                   │                    
       │    55 │   │   else:                               │                    
       │    56 │   │   │   self._summary_time = contextlib │                    
       │    57 │   │                                       │                    
       │ ❱  58 │   │   await self._async_setup_grpc_server │                    
       │    59 │                                           │                    
       │    60 │   async def _async_setup_grpc_server(self │                    
       │    61 │   │   """                                 │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/serve/r… │                    
       │ in _async_setup_grpc_server                       │                    
       │                                                   │                    
       │    65 │   │   # Keep this initialization order    │                    
       │    66 │   │   # otherwise readiness check is not  │                    
       │    67 │   │   # The DataRequestHandler needs to b │                    
       │ ❱  68 │   │   self._data_request_handler = DataRe │                    
       │    69 │   │   │   self.args, self.logger, self.me │                    
       │    70 │   │   )                                   │                    
       │    71                                             │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/serve/r… │                    
       │ in __init__                                       │                    
       │                                                   │                    
       │    41 │   │   self.args.parallel = self.args.shar │                    
       │    42 │   │   self.logger = logger                │                    
       │    43 │   │   self._is_closed = False             │                    
       │ ❱  44 │   │   self._load_executor(metrics_registr │                    
       │    45 │   │   self._init_monitoring(metrics_regis │                    
       │    46 │                                           │                    
       │    47 │   def _init_monitoring(self, metrics_regi │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/serve/r… │                    
       │ in _load_executor                                 │                    
       │                                                   │                    
       │    79 │   │   :param metrics_registry: Optional p │                    
       │       passed to the executor so that it can expos │                    
       │    80 │   │   """                                 │                    
       │    81 │   │   try:                                │                    
       │ ❱  82 │   │   │   self._executor: BaseExecutor =  │                    
       │    83 │   │   │   │   self.args.uses,             │                    
       │    84 │   │   │   │   uses_with=self.args.uses_wi │                    
       │    85 │   │   │   │   uses_metas=self.args.uses_m │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/jaml/__… │                    
       │ in load_config                                    │                    
       │                                                   │                    
       │   727 │   │   │                                   │                    
       │   728 │   │   │   if allow_py_modules:            │                    
       │   729 │   │   │   │   _extra_search_paths = extra │                    
       │ ❱ 730 │   │   │   │   load_py_modules(            │                    
       │   731 │   │   │   │   │   no_tag_yml,             │                    
       │   732 │   │   │   │   │   extra_search_paths=(_ex │                    
       │   733 │   │   │   │   │   if s_path               │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/jaml/he… │                    
       │ in load_py_modules                                │                    
       │                                                   │                    
       │   269 │   _finditem(d)                            │                    
       │   270 │   if mod:                                 │                    
       │   271 │   │   mod = [complete_path(m, extra_searc │                    
       │ ❱ 272 │   │   PathImporter.add_modules(*mod)      │                    
       │   273                                             │                    
       │                                                   │                    
       │ /Users/fengwang/Jina/workspace/jina/jina/importe… │                    
       │ in add_modules                                    │                    
       │                                                   │                    
       │   144 │   │   # assume paths are Python module na │                    
       │   145 │   │   not_python_module_paths = []        │                    
       │   146 │   │   for path in paths:                  │                    
       │ ❱ 147 │   │   │   if not os.path.isfile(path):    │                    
       │   148 │   │   │   │   try:                        │                    
       │   149 │   │   │   │   │   importlib.import_module │                    
       │   150 │   │   │   │   except:                     │                    
       │                                                   │                    
       │ /Users/fengwang/.pyenv/versions/3.8.6/Library/Fr… │                    
       │ in isfile                                         │                    
       │                                                   │                    
       │    27 def isfile(path):                           │                    
       │    28 │   """Test whether a path is a regular fil │                    
       │    29 │   try:                                    │                    
       │ ❱  30 │   │   st = os.stat(path)                  │                    
       │    31 │   except (OSError, ValueError):           │                    
       │    32 │   │   return False                        │                    
       │    33 │   return stat.S_ISREG(st.st_mode)         │                    
       ╰───────────────────────────────────────────────────╯                    
       TypeError: stat: path should be string, bytes,                           
       os.PathLike or integer, not NoneType                                     
ERROR  Flow@26027 Flow is aborted due to ['executor0'] can   [07/15/22 16:09:18]
       not be started.                              
```
- ...
- ...
- [ ] check and update documentation. See [guide](https://github.com/jina-ai/jina/CONTRIBUTING.md#documentation-guidelines) and ask the team.

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
Please we need a much better test for this

**Comment by numb3r3** [comment]
@JoanFM tbh, I have no idea how to produce a better test,  to add an integrated test?

**Comment by JoanFM** [comment]
> @JoanFM tbh, I have no idea how to produce a better test, to add an integrated test?

add an integration test yes, not sure either but we need this

**Review by JoanFM — CHANGES_REQUESTED** [review]
This test does not seem sufficient, I need a differnet module, `BaseExecutor` is something that already works in Jina

**Comment by numb3r3** [comment]
I cannot find any other Executor without introducing additional dependency. That's the reason I also test whether it will work with a wrong `pymodule` path.

**Comment by JoanFM** [comment]
> I cannot find any other Executor without introducing additional dependency. That's the reason I also add a test to guarantee it will fail with a wrong `pymodule` path.

u can have something in our CI/CD to do this test where u install another module. The principle is that we cannot have untested features like this. How will we make sure that we keep supporting this feature if we have nothing to test it

**Comment by numb3r3** [comment]
> u can have something in our CI/CD to do this test where u install another module.

I'm worried this will broke our CI/CD as we cannot ensure another module is available forever.  Anyway, I use `sys.path.append(cur_dir)` to create a new Executor module in our test.

File: tests/unit/serve/executors/test_executor.py [hunk 1]
```diff
@@ -75,6 +77,34 @@ def test_executor_import_with_external_dependencies(capsys):
...
+        jtype: BaseExecutor
+        metas:
+            py_modules:
+                - jina.no_valide.executor
+        '''
+        )
+
+    import sys
+
+    sys.path.append(cur_dir)
```
Comment (JoanFM) [inline tests/unit/serve/executors/test_executor.py:93]: do this with an independent file and in a fixture separated from the test logic

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/unit/serve/executors/test_executor.py (hunk 1 above)
Comment (numb3r3) [reply tests/unit/serve/executors/test_executor.py]: no need anymore, we can directly use the tests folder as the module root.

File: tests/unit/serve/executors/test_executor.py [hunk 2]
```diff
@@ -88,17 +88,14 @@ def test_executor_with_pymodule_path():
...
-    sys.path.append(cur_dir)
     ex = Executor.load_config(
         '''
     jtype: TestExecutor
     with:
         bar: 123
     metas:
         py_modules:
-            - metas_executors
+            - unit.serve.executors.metas_executors
```
Comment (JoanFM) [inline tests/unit/serve/executors/test_executor.py:96]: can we dedicate one file to this, I would not like to add inter tests dependencies

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/unit/serve/executors/test_executor.py (hunk 2 above)
Comment (numb3r3) [reply tests/unit/serve/executors/test_executor.py]: done!

**Review by JoanFM — APPROVED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**